### ISSUE TITLE
Cannot abort attempts to create connections

### ISSUE THREAD

**Issue by acrosman** [SFDO-Tooling/CumulusCI#2027]
**Describe the bug**
I cannot cancel an attempt to create a connection in Windows 10/PowerShell using Ctrl-c.

**Reproduction steps**
1. Run `cci org connect org_name`
2. Web browser launches, and expected messages appear in PowerShell.
3. In powershell hit ctrl-c
4. No response.  Eventually the attempt will time out.

**Context**
Please include the following information at minimum.

- Your CumulusCI version (`cci version`). 
- Your Python version (`python --version`)
```
> cci version
CumulusCI version: 3.18.0 (C:\Users\ahcrosman\.local\bin\cci)
Python version: 3.8.5 (c:\users\ahcrosman\.local\pipx\venvs\cumulusci\scripts\python.exe)
```

- Your operating system and version.
Windows 10

- If using Windows, whether you are using Command Prompt, PowerShell, Bash on Windows, or WSL.
PowerShell

- If possible, whether you installed CumulusCI using `brew`, `pip`, or `pipx`.
pipx

**Comment by davisagli** [comment]
@acrosman We'll try to look into this, but don't have easy access to a Windows environment. Can you check if Ctrl+Break works as an alternative?

**Comment by acrosman** [comment]
As my keyboard lacks a break key, no I cannot.

**Comment by acrosman** [comment]
During the SFDO sprint this week @prescod and I observed the same behavior on another PC's machine.  She also lacked a break key, looks like those are either uncommon or so badly labelled on laptops that no can find them these days.

**Comment by prescod** [comment]
@acrosman Are you able to test the PR #2185  and see if it fixes your problem?

**Comment by prescod** [comment]
@acrosman : If you get a chance, please test these scenarios:

1. Does normal cci org connect work?
2. Does Ctrl-C work?
3. Does the 5 minute timeout work?

**Comment by acrosman** [comment]
Upgraded package cumulusci from 3.20.1 to 3.23.0, then swapped in saleforce.py from the PR.  That led to the new file failing.  Is there another version I need to be working from?  That line isn't from this PR,

```
> cci org connect org_name
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.1776.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.8_3.8.1776.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\ahcrosman\.local\bin\cci.exe\__main__.py", line 4, in <module>
  File "c:\users\ahcrosman\.local\pipx\venvs\cumulusci\lib\site-packages\cumulusci\cli\cci.py", line 31, in <module>
    from cumulusci.core.config import OrgConfig
  File "c:\users\ahcrosman\.local\pipx\venvs\cumulusci\lib\site-packages\cumulusci\core\config\__init__.py", line 23, in <module>
    from cumulusci.core.config.OrgConfig import OrgConfig
  File "c:\users\ahcrosman\.local\pipx\venvs\cumulusci\lib\site-packages\cumulusci\core\config\OrgConfig.py", line 17, in <module>
    from cumulusci.oauth.salesforce import SalesforceOAuth2
  File "c:\users\ahcrosman\.local\pipx\venvs\cumulusci\lib\site-packages\cumulusci\oauth\salesforce.py", line 21, in <module>
    from cumulusci.utils.http.requests_utils import safe_json_from_response
ModuleNotFoundError: No module named 'cumulusci.utils.http'
```

**Comment by prescod** [comment]
Sorry I forgot that I also changed that file in another PR this week.

The file you are missing is:

https://github.com/SFDO-Tooling/CumulusCI/blob/2957ae8769fa715bc1c40459762781f25de0923a/cumulusci/utils/http/requests_utils.py

You could probably just delete the import line in salesforce.py and paste
in the code of the function called safe_json_from_response. You might not
even need to add `from json import JSONDecodeError` but you could do so to
be careful if you like.

**Comment by acrosman** [comment]
Added the other new file, and we appear to be up and running.

Ctrl-c results in the following:
```
> cci org connect org_name
Launching web browser for URL https://login.salesforce.com/services/oauth2/authorize?response_type=code&client_id=3MVG9i1HRpGLXp.or6OVlWVWyn8DXi9xueKNM4npq_AWh.yqswojK9sE5WY7f.biP0w7bNJIENfXc7JMDZGO1&redirect_uri=http://localhost:8080/callback&scope=web%20full%20refresh_token&prompt=login
Spawning HTTP server at http://localhost:8080/callback with timeout of 300 seconds.
If you are unable to log in to Salesforce you can press ctrl+c to kill the server and return to the command line.


Aborted!
```

The actual connection process results in your new graphics.  Although I will offer that when I first loaded this issue tonight one of the links was dead and returning a 404. Giphy as fixed whatever issue they were having but you might think about using images in SFDO's control not a 3rd party.

And I even waited around for the timeout to complete, and it eventually gave up on me.  Of the three options that was the only silent one.  Might be nice to include a message when that happens as well.

**Comment by davisagli** [comment]
Fixed in cumulusci 3.24.0
